### DISCUSSION TITLE
Should renovate obey nuget package restrictions

### DISCUSSION THREAD

**Discussion by phatcher** [renovatebot/renovate#24619] [Request Help] [answered]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.105.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Within a nuget package reference I can constrain the allowable values e.g. 

```
  <ItemGroup>
    <PackageReference Include="Scrutor" Version="[4.1.0]" />      
  </ItemGroup>
```

This limits Scructor to the exact version so within VS it won't offer to update the package.

Basic question is should renovate obey these sort of restrictions in addition to what is proposed via the default.json behaviour

### What you tried so far.

I can see both sides of this one, hence asking the question

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
No, it shouldn't ignore a dependency just because it's pinned to a specific version. But if you wish to disable updating of such cases, you can use `matchCurrentValue` like `/^\[.*\]$/` to match any pinned value and set enabled=false or dependencyDashboardApproval=true

**Comment by phatcher** [comment, marked as answer]
Just to note that this restricting `matchCurrentValue` will not current work, contingent on [#21323](https://github.com/renovatebot/renovate/issues/21323) being resolved
